mutiny-core/src/nodemanager.rs

```diff
@@ -1859,6 +1925,134 @@ impl<S: MutinyStorage> NodeManager<S> {
         Ok(mutiny_channels)
     }
 
+    fn get_scb_key(&self) -> SecretKey {
+        let seed = self.mnemonic.to_seed("");
+        let xprivkey = ExtendedPrivKey::new_master(self.network, &seed).unwrap();
+        let path = DerivationPath::from_str(SCB_ENCRYPTION_KEY_DERIVATION_PATH).unwrap();
+        let context = Secp256k1::new();
+
+        xprivkey.derive_priv(&context, &path).unwrap().private_key
+    }
+
+    /// Creates a static channel backup for all the nodes in the node manager.
+    /// The backup is encrypted with the SCB key.
+    pub async fn create_static_channel_backup(&self) -> Result<EncryptedSCB, MutinyError> {
+        let nodes = self.nodes.lock().await;
+        let mut backups: HashMap<PublicKey, (NodeIndex, StaticChannelBackup)> = HashMap::new();
+        for (_, node) in nodes.iter() {
+            let scb = node.create_static_channel_backup()?;
+            backups.insert(node.pubkey, (node.node_index(), scb));
+        }
+
+        let peers = get_all_peers(&self.storage).unwrap_or_default();
+
+        let peer_connections = peers
+            .into_iter()
+            .filter_map(|(n, p)| p.connection_string.map(|str| (n.as_pubkey().unwrap(), str)))
+            .collect::<HashMap<_, _>>();
+
+        let scb = StaticChannelBackupStorage {
+            backups,
+            peer_connections,
+        };
+
+        // encrypt
+        let encryption_key = self.get_scb_key();
+        let scb = scb.encrypt(&encryption_key);
+        log_debug!(
+            self.logger,
+            "Created SCB with a size of {} bytes",
+            scb.encode().len()
+        );
+        Ok(scb)
+    }
+
+    /// Takes an encrypted static channel backup and recovers the channels from it.
+    /// If the backup is encrypted with a different key than the current key, it will fail.
+    pub async fn recover_from_static_channel_backup(
+        &self,
+        scb: EncryptedSCB,
+    ) -> Result<(), MutinyError> {
+        // decrypt
+        let encryption_key = self.get_scb_key();
+        let scb = scb.decrypt(&encryption_key)?;
+
+        // stop all nodes, todo stop in parallel
+        for node in self.nodes.lock().await.values() {
+            node.stop().await?;
+        }
+
+        for (pubkey, (node_index, backup)) in scb.backups {
+            // find the uuid if we have it, otherwise create a new one and save it
```

Comment:
Should we just store the uuid with the backup?

Reply:
We could but it just increases the size of the encoding and wouldn't really help. Need to do this because on startup we create the first node if we dont have, so when restoring from seed it'll just generate a new uuid anyways and we'd need to use that